Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -12270,8 +12283,127 @@ def get_server_config():
     return server_config
 
 
-from DemistoClassApiModule import *     # type:ignore [no-redef]  # noqa:E402
+def xsoar_profiler(func):
+    """
+    A decorator that profiles the execution of a function and generates a profiling report.
+    To use, decorate the function that calls the function you want to profile with @xsoar_profiler.
+    Example: I want to profile the function_to_profile() function:
+        function_to_profile():
+            # some code
+
+        @xsoar_profiler
+        foo():
+            function_to_profile()
+
+    :param func: The function to be profiled.
+    :return: The profiled function.
+    """
+    import cProfile
+    import threading
+
+    def profiler_wrapper(*args, **kwargs):
+        """
```

Comment:
we should call to the demisto support_multithreading function in the beginning of this decorator

Reply:
I'm not sure we need it as the integration itself doesn't run in multithreaded. The integration code itself runs on the secondary thread.